gnd/src/main/java/com/google/android/gnd/ui/editrecord/EditRecordViewModel.java

```diff
@@ -286,4 +314,36 @@ private boolean hasUnsavedChanges() {
   private boolean hasErrors() {
     return !errors.isEmpty();
   }
+
+  @AutoValue
+  public abstract static class EditRecordRequest {
```

Comment:
This seems like an unfortunate amount of boilerplate.. Can we just use `EditRecordFragmentArgs`? Note that these generated classes are just "structs" w/o and reference to the view.

In future iterations, the view model can already have the required state and then we can just pass the min required state in the stream that triggers the edit.

Reply:
Sure thing